frontend/app/routes/settings/change-password.tsx

```diff
@@ -0,0 +1,180 @@
+import { Button, SubmitButton } from "~/components/ui/button";
+import {
+  FieldSet,
+  FieldSetLabel,
+  FieldSetPasswordToggleInput
+} from "~/components/ui/fieldset";
+import { Separator } from "~/components/ui/separator";
+import { userQueries } from "~/lib/queries";
+import {
+  type ErrorResponseFromAPI,
+  getFormErrorsFromResponseData
+} from "~/lib/utils";
+import { queryClient } from "~/root";
+import { getCsrfTokenHeader, metaTitle } from "~/utils";
+import type { Route } from "./+types/change-password";
+
+export const meta: Route.MetaFunction = () => [metaTitle("Account Settings")];
+
+export async function clientAction({ request }: Route.ClientActionArgs) {
+  const formData = await request.formData();
+  const intent = formData.get("intent")?.toString();
+
+  switch (intent) {
+    case "change_password":
+      return changePassword(formData);
+    default: {
+      throw new Error("Unexpected intent");
+    }
+  }
+}
+
+async function changePassword(formData: FormData) {
+  const credentials = {
+    current_password: formData.get("current_password")!.toString(),
+    new_password: formData.get("new_password")!.toString(),
+    confirm_password: formData.get("confirm_password")!.toString()
+  };
+
+  if (credentials.new_password !== credentials.confirm_password) {
+    return {
+      success: false,
+      errors: {
+        type: "validation_error",
+        errors: [
+          {
+            attr: "confirm_password",
+            code: "validation_error",
+            detail: "Your passwords do not match"
+          }
+        ]
+      } satisfies ErrorResponseFromAPI
+    };
+  }
+
+  const { error: errors, data } = await apiClient.POST(
+    "/api/auth/change-password/",
+    {
+      headers: await getCsrfTokenHeader(),
+      body: credentials
+    }
+  );
+
+  if (errors) return { success: false, errors };
+
+  queryClient.removeQueries(userQueries.authedUser);
+
+  toast.success("Password updated successfully");
+
+  throw redirect("/settings/account");
+}
+
+export default function UserSettingsPage({}: Route.ComponentProps) {
+  return (
+    <section className="flex flex-col gap-4">
+      <div className="flex items-center gap-4">
+        <h2 className="text-2xl">Change Password</h2>
+      </div>
+      <Separator />
+
+      <p className="text-gray">
+        Update your account password. Make sure to use a strong password
+      </p>
+
+      <ChangePassword />
+    </section>
+  );
+}
+
+function ChangePassword() {
+  const navigation = useNavigation();
+  const fetcher = useFetcher<typeof clientAction>();
+  const isPending =
+    navigation.state === "loading" || navigation.state === "submitting";
+  const errors = getFormErrorsFromResponseData(fetcher.data?.errors);
+
+  const formRef = React.useRef<React.ComponentRef<"form">>(null);
+
+  return (
+    <fetcher.Form
+      method="POST"
+      ref={formRef}
+      className="space-y-6 animate-in fade-in duration-300 max-w-lg"
```

Comment:
```suggestion
      className="flex flex-col gap-6 max-w-lg"
```
No animations here, also `flex gap-` instead of `space-y-`